walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,51 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> times = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+    long duration = (long) request.getDuration();
+    if (duration > TimeRange.WHOLE_DAY.duration()){
+      return times;
+    }
+    times.add(TimeRange.WHOLE_DAY);
```

Comment:
IIUC this is saying if there are no attendees in the new meeting request, return all times the whole day as available.  Not necessarily though.  If events had some meetings booked, then not all times would be available.

(I don't think case was considered in the test class, it would be good to account for this though.  Even better if you update the test class to add this test case.

Reply:
I don't understand. If the meeting request has no attendees, then no one's schedules should matter and therefore the entire day should be available.

Reply:
Ok what your saying also makes sense and is definitely a valid approach.  I'm thinking more in terms of no attendees = "probe for all open spaces", in which case it'll try to look up and return all availabilities.